src/main/java/bio/terra/service/filedata/azure/blobstore/AzureBlobStorePdao.java

```diff
@@ -0,0 +1,201 @@
+import bio.terra.service.filedata.FSFileInfo;
+import bio.terra.service.filedata.azure.util.BlobContainerClientFactory;
+import bio.terra.service.filedata.azure.util.BlobCrl;
+import bio.terra.service.filedata.google.firestore.FireStoreFile;
+import bio.terra.service.profile.ProfileDao;
+import bio.terra.service.resourcemanagement.azure.AzureContainerPdao;
+import bio.terra.service.resourcemanagement.azure.AzureResourceConfiguration;
+import bio.terra.service.resourcemanagement.azure.AzureResourceDao;
+import bio.terra.service.resourcemanagement.azure.AzureStorageAccountResource;
+import bio.terra.service.resourcemanagement.azure.AzureStorageAccountResource.ContainerType;
+import com.azure.core.credential.TokenCredential;
+import com.azure.storage.blob.BlobUrlParts;
+import com.azure.storage.blob.models.BlobProperties;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.google.common.annotations.VisibleForTesting;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.Set;
+import java.util.UUID;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AzureBlobStorePdao {
+  private static final Set<String> VALID_TLDS =
+      Set.of("blob.core.windows.net", "dfs.core.windows.net");
+
+  private final ProfileDao profileDao;
+  private final AzureContainerPdao azureContainerPdao;
+  private final AzureResourceConfiguration resourceConfiguration;
+  private final AzureResourceDao azureResourceDao;
+
+  @Autowired
+  public AzureBlobStorePdao(
+      ProfileDao profileDao,
+      AzureContainerPdao azureContainerPdao,
+      AzureResourceConfiguration resourceConfiguration,
+      AzureResourceDao azureResourceDao) {
+    this.profileDao = profileDao;
+    this.azureContainerPdao = azureContainerPdao;
+    this.resourceConfiguration = resourceConfiguration;
+    this.azureResourceDao = azureResourceDao;
+  }
+
+  public FSFileInfo copyFile(
+      FileLoadModel fileLoadModel,
+      String fileId,
+      AzureStorageAccountResource storageAccountResource) {
+
+    BillingProfileModel profileModel =
+        profileDao.getBillingProfileById(fileLoadModel.getProfileId());
+
+    BlobContainerClientFactory destinationClientFactory =
+        getTargetDataClientFactory(profileModel, storageAccountResource, false);
+
+    BlobUrlParts blobUrl = BlobUrlParts.parse(fileLoadModel.getSourcePath());
+    BlobContainerClientFactory sourceClientFactory;
+    if (isSignedUrl(fileLoadModel.getSourcePath())) {
+      sourceClientFactory = getSourceClientFactory(fileLoadModel.getSourcePath());
+    } else {
+      // Use application level authentication
+      sourceClientFactory =
+          getSourceClientFactory(
+              blobUrl.getAccountName(),
+              resourceConfiguration.getAppToken(UUID.fromString(profileModel.getTenantId())),
```

Comment:
Is there a reason why `tennantId` isn't a `UUID`?

Reply:
Nope.  That and subscription should be UUIDs.  I'll update

Reply:
I'm actually going to change this in a different ticket since that's an API change that has a decent reach and I'd like that to be a separable ticket (DR-1964)